pkg/kp/pcstore/consul_store.go

```diff
@@ -154,6 +157,76 @@ func (s *consulStore) Delete(id fields.ID) error {
 	return s.applicator.RemoveAllLabels(labels.PC, id.String())
 }
 
+// performs a safe (ie check-and-set) mutation of the pc with the given id,
+// using the given function
+// if the mutator returns an error, it will be propagated out
+// if the returned PC has id="", then it will be deleted
+func (s *consulStore) mutatePc(
```

Comment:
also you're going to have to name this `MutatePC` otherwise it won't be exported outside this package